xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/SkinEnvironmentResource.java

```diff
@@ -45,18 +56,41 @@ public SkinEnvironmentResource(String path, String resourceName, ResourceReposit
         this.xcontextProvider = xcontextProvider;
     }
 
+    private URLConfiguration getURLConfiguration() {
```

Comment:
I'm actually really not sure about this one in terms of perf: it means that for each resource, we will need to load an URLConfiguration component, is it ok? @tmortagne

Reply:
Why don't you inject it ?

Reply:
Wouldn't it be the same in terms of perf?

Reply:
Sorry I read too fast and did not noticed where you where. You could do the same as what we do for the XWikiContext provider: pass the instance to the constructor.

Reply:
> You could do the same as what we do for the XWikiContext provider: pass the instance to the constructor.

I'll check that.